syncapi/storage/sqlite3/output_room_events_table.go

```diff
@@ -298,16 +269,18 @@ func (s *outputRoomEventsStatements) InsertEvent(
 		return 0, err
 	}
 
-	addStateJSON, err := json.Marshal(addState)
-	if err != nil {
-		return 0, err
+	var addStateJSON, removeStateJSON []byte
+	if len(addState) > 0 {
+		addStateJSON, err = json.Marshal(addState)
+	}
+	if len(removeState) > 0 {
+		removeStateJSON, err = json.Marshal(removeState)
```

Comment:
If `addState` contains invalid JSON but `removeState` does not then we will miss the error because on `:274` we assign to err:
```
addStateJSON, err = json.Marshal(addState)
```
fail to check it, then clobber `err` here. Add a `if err != nil` check please.

Reply:
Done.